# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1305477626 25200
#      Sun May 15 09:40:26 2011 -0700
# Branch stable
# Node ID 2e9cf640379b9361fa1a9701017e251824afb1a9
# Parent  f83ec5ab90adf7a8dc285b0df5624bbe8abe7d57
trapz.m, cumtrapz.m: Quote ':' in initialization statement to prevent it being
interpreted as colon operator.

* trapz.m cumtrapz.m: Quote ':' in initialization statement.

diff --git a/scripts/general/cumtrapz.m b/scripts/general/cumtrapz.m
--- a/scripts/general/cumtrapz.m
+++ b/scripts/general/cumtrapz.m
@@ -78,17 +78,17 @@ function z = cumtrapz (x, y, dim)
   else
     if (!(isscalar (dim) && dim == fix (dim))
         || !(1 <= dim && dim <= nd))
       error ("trapz: DIM must be an integer and a valid dimension");
     endif
   endif
 
   n = sz(dim);
-  idx1 = idx2 = repmat ({:}, [nd, 1]);
+  idx1 = idx2 = repmat ({':'}, [nd, 1]);
   idx1{dim} = 2 : n;
   idx2{dim} = 1 : (n - 1);
 
   if (! have_xy)
     z = 0.5 * cumsum (x(idx1{:}) + x(idx2{:}), dim);
   else
     if (isvector (x) && !isvector (y))
       if (length (x) != sz(dim))
diff --git a/scripts/general/trapz.m b/scripts/general/trapz.m
--- a/scripts/general/trapz.m
+++ b/scripts/general/trapz.m
@@ -92,17 +92,17 @@ function z = trapz (x, y, dim)
   else
     if (!(isscalar (dim) && dim == fix (dim))
         || !(1 <= dim && dim <= nd))
       error ("trapz: DIM must be an integer and a valid dimension");
     endif
   endif
 
   n = sz(dim);
-  idx1 = idx2 = repmat ({:}, [nd, 1]);
+  idx1 = idx2 = repmat ({':'}, [nd, 1]);
   idx1{dim} = 2 : n;
   idx2{dim} = 1 : (n - 1);
 
   if (! have_xy)
     z = 0.5 * sum (x(idx1{:}) + x(idx2{:}), dim);
   else
     if (isvector (x) && !isvector (y))
       if (length (x) != sz(dim))
